mne/io/brainvision/brainvision.py

```diff
@@ -496,13 +496,35 @@ def _get_vhdr_info(vhdr_fname, eog, misc, scale, montage):
                 if hp_s:
                     info['highpass'] = 1. / info['highpass']
         else:
+            heterogeneous_hp_filter = True
             if hp_s:
+                # We convert channels with disabled filters to having
+                # highpass relaxed / no filters
+                highpass = [float(filt) if filt not in ('NaN', 'Off', 'DC')
+                            else np.Inf for filt in highpass]
                 info['highpass'] = np.min(np.array(highpass, dtype=np.float))
+                # Coveniently enough 1 / np.Inf = 0.0, so this works for
+                # DC / no highpass filter
                 info['highpass'] = 1. / info['highpass']
+
+                # not exactly the cleanest use of FP, but this makes us
+                # more conservative in *not* warning.
+                if info['highpass'] == 0.0:
+                    # not actually heterogeneous in effect
+                    # ... just heterogeneously disabled
+                    heterogeneous_hp_filter = False
             else:
+                highpass = [float(filt) if filt not in ('NaN', 'Off', 'DC')
+                            else 0.0 for filt in highpass]
                 info['highpass'] = np.max(np.array(highpass, dtype=np.float))
-            warn('Channels contain different highpass filters. Highest filter '
-                 'setting will be stored.')
+                if info['highpass'] == 0.0:
+                    # not actually heterogeneous in effect
+                    # ... just heterogeneously disabled
+                    heterogeneous_hp_filter = False
+            if heterogeneous_hp_filter:
+                warn('Channels contain different highpass filters. '
+                     'Highest filter setting will be stored.')
```

Comment:
> It's more obvious in the low pass case: a low pass at X destroys information above X, so to be on the safe side, you should inform the user that they should not expect information above X.

This is true, but think about decimation. If I low-pass only half of my channels and decimate, now our subsequent warnings are broken and people can end up with aliasing. To me someone processing data thinking that they *did* low-pass is more dangerous than someone processing data thinking that they *didn't* low-pass. It's also inconsistent with `raw.filter(..., picks=picks)`, which only sets the `info['lowpass']` property if *all* data channels are low-passed, which was designed to avoid this issue.

I'd prefer to change this behavior as a bug/inconsistency fix. @agramfort any opinion?